prov/efa/src/rdm/efa_rdm_ope.c

```diff
@@ -57,7 +57,7 @@ void efa_rdm_txe_construct(struct efa_rdm_ope *txe,
 
 	dlist_init(&txe->queued_pkts);
 
-	if (ep->user_info->mode & FI_MSG_PREFIX)
+	if (txe->iov_count > 0 && ep->user_info->mode & FI_MSG_PREFIX)
```

Comment:
I am worried that this is a hacky workaround. I would audit the code to figure out whether we need to use this function to construct ope for the handshake packet.

Reply:
Are you referring to the change that adds `txe->iov_count > 0` to the condition or the condition as a whole? It's not currently valid to call `ofi_consume_iov_desc()` with an empty IOV - so that change is necessary.

Reply:
I don't want handshake efa_rdm_post_handshake call `efa_rdm_txe_construct`, so you don't have to make this change

Reply:
efa_rdm_txe_construct should be called with a `msg` passed by application. For handshake, we don't have any msg.